test/patch/sw_pg_patch_integrate_test.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2023 SolarWinds, LLC.
+# All rights reserved.
+
+require 'minitest_helper'
+require './lib/solarwinds_apm/config'
+require './lib/solarwinds_apm/opentelemetry'
+require './lib/solarwinds_apm/support/txn_name_manager'
+require './lib/solarwinds_apm/otel_config'
+require './lib/solarwinds_apm/api'
+require './lib/solarwinds_apm/support'
+require './lib/solarwinds_apm/constants'
+require './lib/solarwinds_apm/oboe_init_options'
+require './lib/solarwinds_apm/patch/tag_sql/sw_dbo_utils'
+
+# rubocop:disable Naming/MethodName
+module SolarWindsAPM
+  module Span
+    def self.createSpan(trans_name, domain, span_time, has_error); end
+  end
+end
+# rubocop:enable Naming/MethodName
+
+def pg_dbo_integration_verification(sql, finished_spans)
+  pattern = %r{/\*traceparent='[\da-f-]+'*\*/$}
+  assert_match pattern, finished_spans[0].attributes['sw.query_tag'], "Doesn't match sw.query_tag"
+
+  pattern = %r{^SELECT \* FROM ABC;\s+/\*traceparent='[\da-f-]+'*\*/$}
+  assert_match pattern, sql, "Sql doesn't contain traceparent"
+end
+
+describe 'pg patch integrate test' do
+  puts "\n\033[1m=== TEST RUN PG PATCH TEST: #{RUBY_VERSION} #{File.basename(__FILE__)} #{Time.now.strftime('%Y-%m-%d %H:%M')} ===\033[0m\n"
+
+  let(:sdk) { OpenTelemetry::SDK }
+  let(:exporter) { sdk::Trace::Export::InMemorySpanExporter.new }
+  let(:span_processor) { sdk::Trace::Export::SimpleSpanProcessor.new(exporter) }
+
+  it 'tag_sql_pg_integrate_test' do
+    require './lib/solarwinds_apm/patch/tag_sql/sw_pg_patch'
+
+    OpenTelemetry::SDK.configure(&:use_all)
+    OpenTelemetry.tracer_provider.add_span_processor(span_processor)
+
+    client_ancestors = PG::Connection.ancestors
+    _(client_ancestors[0]).must_equal OpenTelemetry::Instrumentation::PG::Patches::Connection
+    _(client_ancestors[1]).must_equal SolarWindsAPM::Patch::TagSql::SWOPgPatch
+    _(client_ancestors[2]).must_equal PG::Connection
+
+    pg_client = PG::Connection.new
+
+    args = ['SELECT * FROM ABC;', 'a1']
```

Comment:
The second argument `a1` should be an array, see the example in https://deveiate.org/code/pg/PG/Connection.html and https://deveiate.org/code/pg/PG/Connection.html#method-i-exec_params. These params are needed only if placeholder values eg `$1`, `$2` are used in the sql. Some more examples from the `pg` gem: https://github.com/ged/ruby-pg/blob/c6d1047b419e87e57aaec076d9338e80f9bd269a/spec/pg/connection_spec.rb#L1048-L1073 

Pointing this out because it doesn't seem very meaningful to have the parametized sql in the test unless we also intend to verify the collected db attributes to confirm that our patch doesn't interfere with upstream instrumentation handling of these parameters.

Also very curious if there's a way to check our patch does not interfere with the block/returned result. Not sure if it's possible given we're mocking instead of using an actual PG Connection.